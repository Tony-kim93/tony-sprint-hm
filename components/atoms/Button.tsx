import React from 'react';
import * as S from '../../styles/globalStyles';

interface ButtonProps {
  name: string;
  onClick?: any;
  // onClick?: React.MouseEventHandler<HTMLButtonElement> | undefined;
  type?: string;
  style?: any;
}

export default function Button({ name, onClick, type }: ButtonProps) {
  return (
    <S.MainEngineButton type={type} onClick={onClick}>
      {name}
    </S.MainEngineButton>
  );
}
