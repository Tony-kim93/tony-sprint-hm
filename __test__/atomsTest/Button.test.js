import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../../components/atoms/Button';

describe('<BUTTON>', () => {
  it('test every button', () => {
    const name = 'text';
    const onClick = jest.fn();
    const { getByText } = render(<Button name={name} onClick={onClick} />);
    expect(getByText('text')).toBeInTheDocument();
    expect(getByText('text')).toMatchSnapshot();
    fireEvent.click(getByText('text'));
    expect(onClick.mock.calls.length).toBe(1);
  });
});
