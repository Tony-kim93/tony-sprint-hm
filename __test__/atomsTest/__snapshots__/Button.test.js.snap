// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BUTTON> test every button 1`] = `
<button
  class="globalStyles__MainEngineButton-twski7-0 eylqge"
>
  text
</button>
`;
